### PR TITLE
Move RssReader fetch_user to Active Jobs

### PR DESCRIPTION
**Pull request by cyrillefr**
- add job + spec job
  - refactor call

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Move RssReader fetch_user to Active Jobs
## Related Tickets & Documents
#3136 
## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def update
     set_user
     set_tabs(params["user"]["tab"])
     if @user.update(permitted_attributes(@user))
-      RssReader.new.delay.fetch_user(@user) if @user.feed_url.present?
+      RssReaderFetchUserJob.perform_later(@user.id) if @user.feed_url.present?
```
Comment (rhymes) [inline app/controllers/users_controller.rb]: I wonder if the condition `if @user.feed_url.present?` shouldn't be moved inside the job

For two reasons: 
- if it's called in another part of the app, we have to remember to check the same condition
- if between the scheduling and execution the feed_url gets removed, the job will fail

what do you think @lightalloy ?

**Comment by rhymes** [comment]
Thanks @cyrillefr you're doing a lot of awesome work on async jobs!

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (lightalloy) [reply app/controllers/users_controller.rb]: Good point, the second reason seems especially strong to me.

Comment (cyrillefr) [reply app/controllers/users_controller.rb]: I have updated the job in that way.

**Pull request merged**